src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -116,20 +178,111 @@ function performInternalNavigation(absoluteInternalHref: string, interceptedLink
   resetScrollAfterNextBatch();
 
   if (!replace) {
-    history.pushState(state, /* ignored title */ '', absoluteInternalHref);
+    currentHistoryIndex++;
+    history.pushState({
+      userState: state,
+      index: currentHistoryIndex,
```

Comment:
Very minor nit-pick: consider naming this `_index` to just give us a justification for expecting people to know this is an internal implementation detail we might change in the future!